mmv1/third_party/terraform/resources/resource_dataproc_cluster.go.erb

```diff
@@ -13,8 +13,17 @@ import (
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
 
+<% if version == "ga" -%>
+	"google.golang.org/api/dataproc/v1"
+<% else -%>
 	"google.golang.org/api/dataproc/v1beta2"
+<% end -%>
 )
+<% if version == "ga" -%>
+<% client = "NewDataprocClient" -%>
```

Comment:
Instead of switching the client on versions here, would it make sense to change the client returned by `NewDataprocClient` to the appropriate client? That way `goimports` will pick up on the right type (I think) throughout.

Reply:
Then we have to use the beta client throughout the beta provider, which I guess is not a bad thing. Currently we mostly use the GA client except for this cluster resource that has a couple of beta fields available in the beta provider

Reply:
It's more consistent with MMv1, which is nice, and means we can get rid of the ServiceBeta client entirely. If the extra templating is constrained to provider.go/config.go, we could consider the same for GCE and GKE too.

Not required by any means, though! Your call if you want to try and test it out.

Reply:
This ended up being a somewhat significant change, especially as the client's signature for retrieving IAM policies slightly changed between beta and GA. Should be no user-facing changes though